php-frontend/src/main/java/org/sonar/php/parser/TreeFactory.java

```diff
@@ -42,6 +61,84 @@ public ExpressionTree expression(InternalSyntaxToken token) {
    * [ START ] Expression
    */
 
+  public LiteralTree numericLiteral(InternalSyntaxToken token) {
+    return new LiteralTreeImpl(Tree.Kind.NUMERIC_LITERAL, token);
+  }
+
+  public LiteralTree stringLiteral(InternalSyntaxToken token) {
+    return new LiteralTreeImpl(Tree.Kind.STRING_LITERAL, token);
+  }
+
+  public LiteralTree booleanLiteral(InternalSyntaxToken token) {
+    return new LiteralTreeImpl(Tree.Kind.BOOLEAN_LITERAL, token);
+  }
+
+  public LiteralTree nullLiteral(InternalSyntaxToken token) {
+   return new LiteralTreeImpl(Tree.Kind.NULL_LITERAL, token);
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Make this line start at column 5. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AIndentationCheck)